front/lib/models/assistant.ts

```diff
@@ -0,0 +1,241 @@
+      type: DataTypes.STRING,
+      allowNull: false,
+      unique: true,
+    },
+    title: {
+      type: DataTypes.TEXT,
+      allowNull: false,
+    },
+    created: {
+      type: DataTypes.DATE,
+      allowNull: false,
+    },
+    visibility: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      defaultValue: "private",
+    },
+  },
+  {
+    modelName: "assistant_conversation",
+    sequelize: front_sequelize,
+  }
+);
+
+export class AssistantUserMessage extends Model<
+  InferAttributes<AssistantUserMessage>,
+  InferCreationAttributes<AssistantUserMessage>
+> {
+  declare id: number;
+  declare message: string;
+
+  declare userContextUsername: string;
+  declare userContextTimezone: string;
+  declare userContextFullName: string | null;
+  declare userContextEmail: string | null;
+  declare userContextProfilePictureUrl: string | null;
+
+  declare userId: ForeignKey<User["id"]>;
+}
+
+AssistantUserMessage.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    message: {
+      type: DataTypes.TEXT,
+      allowNull: false,
+    },
+    userContextUsername: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    userContextTimezone: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    userContextFullName: {
+      type: DataTypes.STRING,
+      allowNull: true,
+    },
+    userContextEmail: {
+      type: DataTypes.STRING,
+      allowNull: true,
+    },
+    userContextProfilePictureUrl: {
+      type: DataTypes.STRING,
+      allowNull: true,
+    },
+  },
+  {
+    modelName: "assistant_user_message",
+    sequelize: front_sequelize,
+  }
+);
+
+AssistantUserMessage.belongsTo(User, { foreignKey: "userId" });
+User.hasMany(AssistantUserMessage, { foreignKey: "userId" });
+
+export class AssistantAgentMessage extends Model<
+  InferAttributes<AssistantAgentMessage>,
+  InferCreationAttributes<AssistantAgentMessage>
+> {
+  declare id: number;
+  declare message: string | null;
+}
+
+AssistantAgentMessage.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    message: {
+      type: DataTypes.TEXT,
+      allowNull: true,
+    },
+  },
+  {
+    modelName: "assistant_agent_message",
+    sequelize: front_sequelize,
+  }
+);
+
+export class AssistantMessage extends Model<
+  InferAttributes<AssistantMessage>,
+  InferCreationAttributes<AssistantMessage>
+> {
+  declare id: number;
+  declare sId: string;
+
+  declare version: number;
+  declare rank: number;
+  declare status: CreationOptional<"visible" | "deleted">;
+
+  declare assistantConversationId: ForeignKey<AssistantConversation["id"]>;
+  declare parentId: ForeignKey<AssistantMessage["id"]> | null;
+  declare assistantUserMessageId: ForeignKey<AssistantUserMessage["id"]> | null;
+  declare assistantAgentMessageId: ForeignKey<
+    AssistantAgentMessage["id"]
+  > | null;
+}
+
+AssistantMessage.init(
```

Comment:
Should be above next to Conversation

Reply:
As per IRL, it has foreign keys to the other ones so might be clearer if below